packages/react-app/src/views/Vanila/LiquidityProvider.jsx

```diff
@@ -1,129 +1,75 @@
-import { useUserProvider, useContractLoader, useGasPrice, useExternalContractLoader, useContractReader } from '../../hooks'
-import { useUserAddress } from "eth-hooks";
 import WalletConnectProvider from "@walletconnect/web3-provider";
 import { Web3Provider } from "@ethersproject/providers";
 import Web3Modal from "web3modal";
-import { Transactor } from "./../../helpers";
-import { Button, Card, DatePicker, Divider, Input, List, Progress, Slider, Spin, Switch } from "antd";
-import { utils } from 'ethers'
-
-const { parseUnits, formatUnits } = utils
+import { useUserProvider } from "../../hooks";
+import { INFURA_ID, NETWORKS } from "../../constants";
+import { Pool } from "./Pool";
 
 const { ethers } = require("ethers");
 
 const targetNetwork = NETWORKS.localhost;
-const localProviderUrl = targetNetwork.rpcUrl;
-const localProviderUrlFromEnv = process.env.REACT_APP_PROVIDER ? process.env.REACT_APP_PROVIDER : localProviderUrl;
-const localProvider = new ethers.providers.StaticJsonRpcProvider(localProviderUrlFromEnv);
-const blockExplorer = targetNetwork.blockExplorer;
-
-export const LiquidityProvider = () => {
-
-  const [injectedProvider, setInjectedProvider] = useProvider()
-  const userProvider = useUserProvider(injectedProvider, localProvider)
-
-  return <div>
-    <h1>Liquidity Provider</h1>
-    <div>
-      <h2>DAI</h2>
-      <table>
-        <tr>
-          <td><Wallet userProvider={userProvider} /></td>
-          <td><Deposit userProvider={userProvider} /></td>
-        </tr>
-        <tr>
-          <td><Pool userProvider={userProvider} /></td>
-          <td><Button>WITHDRAW</Button></td>
-          <td><Input /></td>
-        </tr>
-      </table>
-    </div>
-  </div>
 
-}
+const useMyUserProvider = () => {
+  const [web3Modal, setWeb3Modal] = useState();
+  const [localProvider, setLocalProvider] = useState();
+  const [injectedProvider, setInjectedProvider] = useState();
 
-const web3Modal = new Web3Modal({
-  // network: "mainnet", // optional
-  cacheProvider: true, // optional
-  providerOptions: {
-    walletconnect: {
-      package: WalletConnectProvider, // required
-      options: {
-        infuraId: INFURA_ID,
-      },
-    },
-  },
-})
-
-const useProvider = () => {
+  useEffect(() => {
+    const localProviderUrlFromEnv = process.env.REACT_APP_PROVIDER
+      ? process.env.REACT_APP_PROVIDER
+      : targetNetwork.rpcUrl;
+    setLocalProvider(new ethers.providers.StaticJsonRpcProvider(localProviderUrlFromEnv));
+  }, []);
 
-  const [injectedProvider, setInjectedProvider] = useState()
+  useEffect(() => {
+    setWeb3Modal(
+      new Web3Modal({
+        // network: "mainnet", // optional
+        cacheProvider: true, // optional
+        providerOptions: {
+          walletconnect: {
+            package: WalletConnectProvider, // required
+            options: {
+              infuraId: INFURA_ID,
+            },
+          },
+        },
+      }),
+    );
+  }, []);
```

Comment:
This is fine for now, but just an fyi I've refactored all this stuff to achieve the same goal but in a "nicer" way. We can consolidate the code later. Looks pretty good tho 👍

Reply:
sure, I'm not frontend dev, so yes to everything you say 😅 

I just wanted to clean a bit the code to make easier to perform that refactor